includes/main.js

```diff
@@ -28,39 +28,53 @@ function initiateApp(){
 }
 function makeGallery(imageArray){
 	//use loops and jquery dom creation to make the html structure inside the #gallery section
-
 	//create a loop to go through the images in the imageArray
-		//create the elements needed for each picture, store the elements in variable
+		for (var imageArrayIndex = 0; imageArrayIndex < imageArray.length; imageArrayIndex++){
+			//$("#gallery").append(imageArray); //css class to append images
 
-		//attach a click handler to the figure you create.  call the "displayImage" function.  
-
-		//append the element to the #gallery section
-	
+		//create the elements needed for each picture, store the elements in variable
+			var fig = $('<figure>', {
+				class: "imageGallery col-xs-12 col-sm-6 col-md-4",
+				style: "background-image:url('"+ imageArray[imageArrayIndex]+"')"
+			});
+
+			var caption = $("figcaption").slice('images/');
+			//attach a click handler to the figure you create.  call the "displayImage" function.
+			fig.click(displayImage);
+			//append the element to the #gallery section
+			$("#gallery").append(fig);
 	// side note: make sure to remove the hard coded html in the index.html when you are done!
-
+	}
 }
 
+
 function addModalCloseHandler(){
 	//add a click handler to the img element in the image modal.  When the element is clicked, close the modal
-	//for more info, check here: https://www.w3schools.com/bootstrap/bootstrap_ref_js_modal.asp	
+	$("#galleryModal").click(function(){
+		$("#galleryModal").modal("hide")
+	});
+	//for more info, check here: https://www.w3schools.com/bootstrap/bootstrap_ref_js_modal.asp
 }
 
-function displayImage(){
+	function displayImage(){
 	//find the url of the image by grabbing the background-image source, store it in a variable
+
+	var backgroundImage = $(this).css('background-image');
+	console.log(backgroundImage);
 	//grab the direct url of the image by getting rid of the other pieces you don't need
 
 	//grab the name from the file url, ie the part without the path.  so "images/pexels-photo-132037.jpeg" would become
 		// pexels-photo-132037
-		//take a look at the lastIndexOf method
+		//take a look at the lastIndexOf method (right side look left)
+	var x = backgroundImage.lastIndexOf('/')
```

Comment:
these are very poor variable names because they don't describe what they are for at all